docs/tutorials/python/tutorial_scripts/wiki.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python3
+"""
+Tutorial script demonstrating the Synapse Wiki models functionality.
+
+This script shows how to:
+1. Create, read, and update wiki pages
+2. Work with WikiPage Markdown
+3. Work with WikiPage Attachments
+4. Work with WikiHeader
+5. Work with WikiHistorySnapshot
+6. Work with WikiOrderHint
+7. Delete wiki pages
+"""
+import gzip
+import os
+import uuid
+
+from synapseclient import Synapse
+from synapseclient.models import (
+    Project,
+    WikiHeader,
+    WikiHistorySnapshot,
+    WikiOrderHint,
+    WikiPage,
+)
+
+# Initialize Synapse client
+syn = Synapse()
+syn.login()
+
+# Create a Synapse Project to work with
+my_test_project = Project(
+    name=f"My Test Project_{uuid.uuid4()}",
+    description="This is a test project for the wiki tutorial.",
+).store()
+print(f"Created project: {my_test_project.name} with ID: {my_test_project.id}")
+
+# Section1: Create, read, and update wiki pages
+# Create a new wiki page for the project with plain text markdown
+root_wiki_page = WikiPage(
+    owner_id=my_test_project.id,
+    title="My Root Wiki Page",
+    markdown="# Welcome to My Root Wiki\n\nThis is a sample root wiki page created with the Synapse client.",
+).store()
+
+# OR you can create a wiki page with an existing markdown file. More instructions can be found in section 2.
+markdown_file_path = "path/to/your_markdown_file.md"
+root_wiki_page = WikiPage(
+    owner_id=my_test_project.id,
+    title="My First Root Wiki Page Version with existing markdown file",
+    markdown=markdown_file_path,
+).store()
+
+# Create a new wiki page with updated content
+root_wiki_page_new = WikiPage(
+    owner_id=my_test_project.id,
+    title="My First Root Wiki Page NEW",
+    markdown="# Welcome to My Root Wiki NEW\n\nThis is a sample root wiki page created with the Synapse client.",
+    id=root_wiki_page.id,
+).store()
+
+# Restore the wiki page to the original version
+wiki_page_restored = WikiPage(
+    owner_id=my_test_project.id, id=root_wiki_page.id, wiki_version="0"
+).restore()
+
+# check if the content is restored
+comparisons = [
```

Comment:
It is probably easier and cleaner to use assert statements here. Like


`assert condition, "Message for failure"`

That was if it does fail, it's clear which one failed the compare check